src/chromo/decay_handler.py

```diff
@@ -0,0 +1,204 @@
+import chromo
+import numpy as np
+
+
+class Pythia8DecayHandler:
+    def __init__(
+        self,
+        seed=None,
+        extra_stable_pids=None,
+        extra_decaying_pids=None,
+        stable_pids=None,
+        decaying_pids=None,
+    ):
+        """
+        Initialize the Pythia8DecayHandler with Pythia8 and set particle types for decay.
+
+        Parameters:
+            seed (int): Random seed for Pythia8 initialization.
+            extra_stable_pids (list[int]): List of additional PDG IDs
+            for stable particles.
+            extra_decaying_pids (list[int]): List of additional PDG IDs
+            for decaying particles.
+            stable_pids (list[int]): List of PDG IDs
+            for exclusively stable particles (all other are decaying if they can decay).
+            decaying_pids (list[int]): List of PDG IDs
+            for exclusively decaying particles (all other are stable).
+
```

Comment:
This is confusing and way too complicated. I don't understand what you mean by "extra" and "exclusive" decay later. This needs to be simplified.

What does this Handler do? I thought it decays particles which the model does not decay. Do you also want to use it to do the opposite, remove daughters of particles which were decayed in the original model? I think this case never happens, or does it?

I assume that we only need to decay particles which have not been decayed by the model. In that case, you only need one argument in addition to seed, the pids of particles which the handler should decay.

Reply:
> Do you also want to use it to do the opposite, remove daughters of particles which were decayed in the original model? 

I am not sure that I understood this. 

I just wanted to have fine-grained control to what is decaying and what is stable. For current needs it is, perhaps, overkill. In current implementation only stable particles should be set. All other will be decaying.

Reply:
Then please don't design something that is more general than what we need. This feature already adds so much complexity to our code. We need to try to keep the complexity minimal.